subprojects/maven/src/main/java/org/gradle/api/publish/maven/internal/publication/DefaultMavenPublication.java

```diff
@@ -126,13 +118,8 @@ public DefaultMavenPublication(
         this.versionMappingStrategy = versionMappingStrategy;
         this.taskDependencyFactory = taskDependencyFactory;
 
-        MavenComponentParser mavenComponentParser = new MavenComponentParser(
-            platformSupport,
-            versionRangeMapper,
-            documentationRegistry,
-            versionMappingStrategy,
-            projectDependencyResolver,
-            mavenArtifactParser
+        MavenComponentParser mavenComponentParser = objectFactory.newInstance(
```

Comment:
🤔 could this be pulled up as a service?

Reply:
See https://github.com/gradle/gradle/pull/26356#discussion_r1331868452